tests/rag_answers/custom_deepeval/metrics/context_relevancy/test_context_relevancy.py

```diff
@@ -0,0 +1,342 @@
+import pytest
+import json
+from unittest.mock import Mock, AsyncMock, patch
+
+from deepeval.test_case import LLMTestCase
+from deepeval.models import GPTModel, DeepEvalBaseLLM
+from deepeval.errors import MissingTestCaseParamsError
+from deepeval.utils import prettify_list
+import re
+
+from govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.context_relevancy import (
+    ContextRelevancyMetric,
+)
+from govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.context_relevancy.schema import (
+    Truth,
+    TruthCollection,
+    InformationNeedsCollection,
+    Verdict,
+    VerdictCollection,
+    ScoreReason,
+)
+from govuk_chat_evaluation.rag_answers.data_models import StructuredContext
+
+
+@pytest.fixture
+def mock_native_model(
+    sample_truths, sample_information_needs, sample_verdicts, sample_reason
+):
+    mock = Mock(spec=GPTModel)
+    mock.get_model_name.return_value = "eu.amazon.nova-pro-v1:0"
+    mock.a_generate = AsyncMock()
+    mock.a_generate.side_effect = [
+        (sample_truths, 0.1),
+        (sample_information_needs, 0.1),
+        (sample_verdicts, 0.1),
+        (sample_reason, 0.1),
+    ]
+    return mock
+
+
+@pytest.fixture
+def mock_non_native_model(
+    sample_truths, sample_information_needs, sample_verdicts, sample_reason
+):
+    mock = Mock(spec=DeepEvalBaseLLM)
+    mock.get_model_name.return_value = "non-native-model"
+    mock.a_generate = AsyncMock()
+    mock.a_generate.side_effect = [
+        (sample_truths, 0.1),
+        (sample_information_needs, 0.1),
+        (sample_verdicts, 0.1),
+        (sample_reason, 0.1),
+    ]
+    return mock
+
+
+@pytest.fixture
+def test_case():
```

Comment:
```suggestion
def mock_test_case():
```

Not sure if this is this causes any warnings or not as pytest expects methods named test_ to be tests. Perhaps the fixture decorator resolves it but I think renaming makes it's intent clearer anyway.